p1-staking-pool-dyn/src/lib.rs

```diff
@@ -275,52 +261,42 @@ impl Contract {
     }
 
     #[private]
-    pub fn mint_callback(&mut self, user: AccountId, amount: U128, close: bool) {
-        env_log!(
-            "mint_callback, env::promise_results_count()={}",
-            env::promise_results_count()
-        );
+    pub fn mint_callback(&mut self, user: AccountId, amount: U128) {
         // after the async call to mint rewards for the user
         assert_eq!(
             env::promise_results_count(),
             1,
             "{}",
             ERR25_WITHDRAW_CALLBACK
         );
+        // check prev promise result
         match env::promise_result(0) {
             PromiseResult::NotReady => unreachable!(),
 
             PromiseResult::Successful(_) => {
                 env_log!("cheddar rewards withdrew {}", amount.0);
                 self.total_rewards += amount.0;
-                if close {
-                    // AUDIT: TODO: Should check that the vault doesn't have `.staked > 0`, because
-                    //    in case of weird async race conditions, the vault might get new staked balance.
-                    self.vaults.remove(&user);
-                    env::log(b"account closed");
```

Comment:
please keep the log. Logs are useful and can be queried in UI.

Reply:
since we renamed the fn, we don't know at this point if we're closing the account. that's why it was removed.
The log is now at pub fn close